### ISSUE TITLE
Jinja directive causing mis-indentation of the rest of the query

### ISSUE THREAD

**Issue by dmohns** [sqlfluff/sqlfluff#4064]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

After the recent release of version 1.4.2 I noticed a lot of queries getting inserted intends in unexpected places, see example below. It seems that the Jinja `{% if ... %}` somehow affects the remainder of the query.

### Expected Behaviour

Indentations around Jinja blocks **should not** affect rest of the query.

```sql
SELECT 1
WHERE TRUE
    {% if true %}
        2
    {% endif %}
ORDER BY 1
;

INSERT INTO table_1 (column_1)
VALUES ('a_value')
;
```

### Observed Behaviour

Indentations around Jinja blocks **affects** rest of the query.

```sql
SELECT 1
WHERE TRUE
    {% if true %}
        2
    {% endif %}
    ORDER BY 1;

    INSERT INTO table_1 (column_1)
    VALUES ('a_value');
```

(Note how everything seems indented after `{% endif %}`).

### How to reproduce

```sql
SELECT 1
WHERE TRUE
    {% if true %}
        2
    {% endif %}
ORDER BY 1
;

INSERT INTO table_1 (column_1)
VALUES ('a_value');
```

https://online.sqlfluff.com/fluffed?sql=H4sIALP2cWMC_wt29XF1DlEw5OUK93ANclUICQp15eVSAIJqVYXMNIWSotJUBdVaiBAIGMFlU_NSgApAcv5BLq5BCk6RIGOsebl4uTz9gl2DQhQ8_UL8FUoSk3JS4w0VNJLzc0pz8-INNXm5whx9Ql2DFTTUE-PLEnNKU9U1gdoAxF2XJosAAAA%3D&dialect=ansi

### Dialect

ansi

### Version

1.4.2 (tested on webversion)

### Configuration

default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
@dmohns - I believe this is now resolved with #3942 . I've just confirmed that in my local env on `main`. Alpha release of 2.0.0 should be coming soon which will allow you to test this.

**Comment by dmohns** [comment]
Great to hear! Looking very much forward to the alpha release.
